pkg/apis/org/v1/backuprestorecommon_types.go

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package v1
+
+const (
+	USERNAME_SECRET_KEY              = "username"
+	PASSWORD_SECRET_KEY              = "password"
+	RESTIC_REPO_PASSWORD_SECRET_KEY  = "repo-password"
+	SSH_PRIVATE_KEY_SECRET_KEY       = "ssh-privatekey"
+	AWS_ACCESS_KEY_ID_SECRET_KEY     = "awsAccessKeyId"
+	AWS_SECRET_ACCESS_KEY_SECRET_KEY = "awsSecretAccessKey"
+)
+
+// +k8s:openapi-gen=true
+// List of supported backup servers
+type BackupServers struct {
+	// Rest server within the cluster.
+	// The server and configuration are created by operator when AutoconfigureRestBackupServer is true
+	Internal RestServerConfig `json:"internal,omitempty"`
+	// Sftp backup server configuration
+	Sftp SftpServerConfing `json:"sftp,omitempty"`
+	// Rest backup server configuration
+	Rest RestServerConfig `json:"rest,omitempty"`
+	// Amazon S3 or alternatives
+	AwsS3 AwsS3ServerConfig `json:"awss3,omitempty"`
+}
+
+// Holds restic repository password to decrypt its content
+// At least one of the fields has to be filled in
+type RepoPassword struct {
+	// Password for restic repository
+	Password string `json:"password,omitempty"`
+	// Secret with 'repo-password' filed
+	PasswordSecretRef string `json:"passwordSecretRef,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+// SFTP backup server configuration
+// Example: user@host://srv/repo
+// Mandatory fields are: RepoPassword, Hostname, Repo, Username, SshKeySecretRef
+type SftpServerConfing struct {
+	// Restic repository password
+	RepoPassword `json:"repoPassword,omitempty"`
+	// Backup server host
+	Hostname string `json:"hostname,omitempty"`
+	// Backup server port
+	Port int `json:"port,omitempty"`
+	// Restic repository path, relative or absolute, e.g. /srv/repo
+	Repo string `json:"repo,omitempty"`
+	// User login on the remote server
+	Username string `json:"username,omitempty"`
+	// Private ssh key under 'ssh-privatekey' field for passwordless login
+	SshKeySecretRef string `json:"sshKeySecretRef,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+// REST backup server configuration
+// Example: https://user:password@host:5000/repo/
+// Mandatory fields are: RepoPassword, Hostname
```

Comment:
this info does not seem to be populated to JSON Schema. I'm not sure if it's possible but would be ideal to get right required field in JSON Schema

Reply:
Probably that's not possible, but I've added duplicate commentary to the fields, so they will be propagated to CRDs yamls.